File: Problem_set_1/Introductory_problems/p04_Increasing_Array/IncreasingArray.java
import java.util.Scanner;

public class IncreasingArray {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        int size_array = scan.nextInt();

        int[] num_arr = new int[(int)size_array];
        
        for(int i = 0; i < num_arr.length; i++){
            num_arr[i] = scan.nextInt();
        }
        
        long totalMoves = 0;
        for(int i = 1; i < num_arr.length; i++){
            int current = num_arr[i];
            int previous = num_arr[i - 1];
            
            if(current < previous){
                long gap = previous - current;
                totalMoves += gap;
                num_arr[i] = previous;
            }
        }
        System.out.println(totalMoves);
        scan.close();
    }
}

File: Problem_set_1/Introductory_problems/p04_Increasing_Array/IncreasingArray.cpp
#include<iostream>
#include<vector>

using namespace std;

int main(){
    int size_array;
    cin >> size_array;

    vector<int> num_arr(size_array);

    for(int i = 0; i < num_arr.size(); i++){
        cin >> num_arr[i];
    }

    long long totalMoves = 0;
    for(int i = 1; i < num_arr.size(); i++){
        int current = num_arr[i];
        int previous = num_arr[i - 1];

        if(current < previous){
            long long gap = previous - current;
            totalMoves += gap;
            num_arr[i] = previous;
        }
    }

    cout << totalMoves << endl;
    return 0;
}